src/scripts/dungeons/DungeonRunner.ts

```diff
@@ -63,7 +65,17 @@ class DungeonRunner {
         }
         if (this.map.playerMoved()) {
             this.timeLeft(this.timeLeft() - GameConstants.DUNGEON_TICK);
-            this.timeLeftPercentage(Math.floor(this.timeLeft() / GameConstants.DUNGEON_TIME * 100));
+            this.timeLeftPercentage(Math.floor(this.timeLeft() / (GameConstants.DUNGEON_TIME * FluteEffectRunner.getFluteMultiplier(GameConstants.FluteItemType.Time_Flute)) * 100));
+        }
+        const currentFluteBonus = FluteEffectRunner.getFluteMultiplier(GameConstants.FluteItemType.Time_Flute);
+        if (currentFluteBonus != this.timeBonus()) {
+            if (currentFluteBonus > this.timeBonus()) {
+                this.timeBonus(currentFluteBonus);
+                this.timeLeft(this.timeLeft() * this.timeBonus());
```

Comment:
Right now, if you get an achievement while completing a dungeon, it will multiply timeLeft again. Maybe you should also check if currentFluteBonus is 1?
And this code could also use a comment or two explaining why it's here.